### ISSUE TITLE
Component and Entity pooling

### ISSUE THREAD

**Issue by apotapov** [junkdog/artemis-odb#16]
I seem to have trouble posting on the Artemis forum, so I figured I'd ping you here instead. I'm the guy working on gdx-artemis. (https://github.com/apotapov/gdx-artemis)

You guys have done an awesome job cleaning up Artemis, and it seems our intentions are pretty well aligned: improving the implementation and making it more peformant, specifically in memory management. Ultimately, I would like to see our codebases converge, rather than diverge maintaining separate forks.

I'm not particularly married to the idea of having libgdx as a dependency for artemis. As a matter of fact in hindsight it's probably a bad idea to do so.

My biggest issue with Artemis has been the lack of entity and component pooling. It wouldn't be so bad if I could instantiate my own Entity and Component managers into an Artemis world, but the framework seems to be hell bent on preventing any reasonable extension of classes. (almost everything internal is either private or package protected)

So, long story short, my proposal is to work with you to add pooling to Artemis. Thoughts?
